components/core/src/clp_s/ArchiveReader.hpp

```diff
@@ -54,18 +104,42 @@ class ArchiveReader {
      */
     void close();
 
+    /**
+     * @return The schema ids in the archive.
+     */
+    [[nodiscard]] std::vector<int32_t> const& get_schema_ids() const { return m_schema_ids; }
+
 private:
+    /**
+     * Appends a set of columns to the givenschema reader.
+     * @param reader
+     * @param extract_timestamp
+     */
+    void append_reader_columns(std::unique_ptr<SchemaReader>& reader, bool extract_timestamp);
+
+    /**
+     * Appends a column to the schema reader.
+     * @param reader
+     * @param column_id
+     */
+    BaseColumnReader*
+    append_reader_column(std::unique_ptr<SchemaReader>& reader, int32_t column_id);
+
+    bool m_is_open;
     std::string m_archive_path;
 
     std::shared_ptr<VariableDictionaryReader> m_var_dict;
     std::shared_ptr<LogTypeDictionaryReader> m_log_dict;
     std::shared_ptr<LogTypeDictionaryReader> m_array_dict;
+    std::shared_ptr<TimestampDictionaryReader> m_timestamp_dict;
 
     std::shared_ptr<SchemaTree> m_schema_tree;
-    ReaderUtils::SchemaMap m_schema_map;
-    std::map<int32_t, SchemaReader*> m_schema_id_to_reader;
+    std::shared_ptr<ReaderUtils::SchemaMap> m_schema_map;
+    std::vector<int32_t> m_schema_ids;
+    std::map<int32_t, SchemaReader::TableMetadata> m_id_to_table_metadata;
 
-    std::shared_ptr<TimestampDictionaryReader> m_timestamp_dict;
+    FileReader m_table_file_reader, m_metadata_file_reader;
+    ZstdDecompressor m_table_decompressor, m_metadata_decompressor;
```

Comment:
```suggestion
    ZstdDecompressor m_table_decompressor;
    ZstdDecompressor m_metadata_decompressor;
```
Or switch to a single ZstdDecompressor per other comment